SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -327,6 +335,30 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 		return;
 	}
 
+	// If the received actor is torn off, don't bother receiving it.
+	// Check the pending add components, to find the root component for the received entity.
+	for (PendingAddComponentWrapper& PendingAddComponent : PendingAddComponents)
+	{
+		if (PendingAddComponent.EntityId != EntityId
+			|| ClassInfoManager->GetCategoryByComponentId(PendingAddComponent.ComponentId) != SCHEMA_Data)
+		{
+			continue;
+		}
+		uint32 componentOffset = 0;
+		if (!ClassInfoManager->GetOffsetByComponentId(PendingAddComponent.ComponentId, componentOffset) || componentOffset != 0)
+		{
+			continue;
+		}
+
+		Worker_ComponentData* ComponentData = static_cast<improbable::DynamicComponent*>(PendingAddComponent.Data.Get())->Data;
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData->schema_type);
+		if (Schema_GetBool(ComponentObject, SpatialConstants::ACTOR_TEAROFF_ID))
```

Comment:
Can we add a sanity check in SchemaGenerator that the tearoff property actually matches this ID? (similar to `ACTOR_COMPONENT_REPLICATES_ID`)

Reply:
Will look into this 👍